Documentation/specs/corehost.md

```diff
@@ -0,0 +1,117 @@
+# corehost runtime/assembly resolution
+
+The shared host locates assemblies and native libraries using a combination of: Servicing Index, Files in the application folder (aka "app-local") and files from package caches.
+
+## Definitions and Formats
+
+### Terms/Notes
+
+* The term "Library" (Title Case) is used throughout this document to refer to a NuGet Package. We use this term because it is used in the code to represent multiple things: Packages, Projects and Framework Assemblies are all types of "Library".
+* All of the assembly resolution here refers to setting up the default assembly load context for the runtime. Further dynamic loads (plugins, etc.) can still have custom resolution logic provided by using an `AssemblyLoadContext` in the managed code. Essentially, we are setting up the necessary assemblies to launch the `Program.Main` (and all the assemblies that are statically-referenced by that).
+
+### Servicing Index
+
+The servicing index is loaded when the `DOTNET_SERVICING` environment variable is non-empty. When this variable is non-empty, it points to a directory that will be the **Servicing Root**. There may be platform-specific default locations, to be determined later.
+
+An index file is located at the path defined by `$DOTNET_SERVICING/servicing_index.txt`. In this file are a series of lines of one of the following formats:
+
+```
+# Lines starting with a '#' and blank lines are ignored.
+
+# Identifies an asset from a NuGet Package that has been serviced.
+package|[Package ID]|[Package Version]|[Original Asset Relative Path]=[New Asset Path, relative to Servicing Root]
+package|System.Threading.Thread|1.2.3.4|lib/dotnet5.4/System.Threading.Thread.dll=patches/abc123/System.Threading.Thread.dll
+
+# TBD: Host/Runtime servicing entries.
+```
+
+Paths in this file are **always** specified using `/`, even on Windows. They must be converted to platform-specific directory separators.
+
+This index is loaded when needed during the Resolution Process (see below).
+
+### Dependencies File
+
+The `.deps` file is a file located alongside the native host. It is stored in a CSV format, with header, of the following format:
+
+```
+LibraryType,LibraryName,LibraryVersion,LibraryHash,AssetType,AssetName,AssetRelativePath
+"Package","Microsoft.CodeAnalysis.Common","1.2.0-beta-20151117-04","sha512-4O3f1iRswbWDz70AGTk0tBA1ivslJUuq37FSx4d0+ZuoBFxNO4oY7ReJgndOJYefaxeTMsCVHSVHJMYOJJphtQ==","runtime","Microsoft.CodeAnalysis","lib\\portable-net45+win8\\Microsoft.CodeAnalysis.dll",
+```
```

Comment:
It's an escape sequence (in the CSV file). Either way, this should be `/` since we always use Unix-style paths in these files and normalize to platform-specific paths later.